components/unarchive_channel_modal/unarchive_channel_modal.jsx

```diff
@@ -0,0 +1,126 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import PropTypes from 'prop-types';
+import React from 'react';
+import {Modal} from 'react-bootstrap';
+import {FormattedMessage} from 'react-intl';
+
+import Constants from 'utils/constants';
+import FormattedMarkdownMessage from 'components/formatted_markdown_message';
+
+export default class UnarchiveChannelModal extends React.PureComponent {
+    static propTypes = {
+
+        /**
+        * Function called when modal is dismissed
+        */
+        onHide: PropTypes.func.isRequired,
+
+        /**
+         * channel data
+         */
+        channel: PropTypes.object.isRequired,
+
+        canViewArchivedChannels: PropTypes.bool,
+
+        actions: PropTypes.shape({
+
+            /**
+            * Function called for deleting channel,
+            */
+
+            unarchiveChannel: PropTypes.func.isRequired,
+        }),
+    }
+
+    constructor(props) {
+        super(props);
+
+        this.state = {show: true};
+    }
+
+    handleUnarchive = () => {
+        if (this.props.channel.id.length !== Constants.CHANNEL_ID_LENGTH) {
+            return;
+        }
+        this.props.actions.unarchiveChannel(this.props.channel.id);
+        this.onHide();
+    }
+
+    onHide = () => {
+        this.setState({show: false});
+    }
+
+    render() {
+        const {canViewArchivedChannels} = this.props;
+        return (
+            <Modal
+                dialogClassName='a11y__modal'
+                show={this.state.show}
+                onHide={this.onHide}
+                onExited={this.props.onHide}
+                role='dialog'
+                aria-labelledby='unarchiveChannelModalLabel'
+                id='unarchiveChannelModal'
+            >
+                <Modal.Header closeButton={true}>
+                    <Modal.Title
+                        componentClass='h1'
+                        id='unarchiveChannelModalLabel'
+                    >
+                        <FormattedMessage
+                            id='unarchive_channel.confirm'
+                            defaultMessage='Confirm UNARCHIVE Channel'
+                        />
+                    </Modal.Title>
+                </Modal.Header>
+                <Modal.Body>
+                    <div className='alert alert-danger'>
+                        {!canViewArchivedChannels &&
+                            <FormattedMarkdownMessage
+                                id='unarchive_channel.question'
+                                defaultMessage='This will unarchive the channel from the team and make its contents accessible for all users. \n \nAre you sure you wish to unarchive the **{display_name}** channel?'
```

Comment:
@adamjclarkson This first sentence feels a bit awkward since "from the team" doesn't fit and making its content accessible is kind of the point of the feature. Maybe we remove it and just keep the second sentence?
```suggestion
                                defaultMessage='Are you sure you wish to unarchive the **{display_name}** channel?'
```

Reply:
Yep, I agree the first sentence doesn't add much.

Reply:
deleted the first sentence